gen/alt.ml

```diff
@@ -360,11 +360,34 @@ module Make(C:Builder.S)
 
     let can_prefix prefix = mk_can_prefix prefix
 
-    let call_rec prefix f0 safes po_safe over n r suff f_rec k =
+    let rec check2 l rl =
+      match l,rl with
+      | _::_,[] | [],_::_ | [],[] ->  false (* should pass a warning, this case shouldn't be possible*)
+      | h::_, hr::[] -> if h=hr then begin false end else true
+      | _::[], _::_ -> true (* no wraparound implemented as cycle is still being built*)
+      | h::t, hr::tr -> if h = hr then begin check2 t tr end else true
+
+
+    let check_cycle rsuff rl =
+      let rsuff = List.map (fun (_,rr) -> rr) rsuff in
+      let rsuff = List.concat rsuff in
+      let r_first = List.hd rsuff in
+      let rec f rej= match rej with
+        | h::t -> if (List.hd h) = r_first then begin
+          let truth = check2 (List.tl rsuff) (List.tl h) in
+          if truth then f t else false
+          end
+          else f t
+        | [] -> true in
+      f rl
```

Comment:
Given the above definition, I think you can do this to check the suffix across all elements in rl
```suggestion
      not (List.exists (fun rl -> is_suffix rl rsuff) rl)
```